Applications/Utils/MeshEdit/UnityPreprocessing.cpp

```diff
@@ -0,0 +1,245 @@
+            new_elems.push_back(createElement<MeshLib::Tri>(*elem, new_nodes, node_map));
+        else if (elem->getGeomType() == MeshLib::MeshElemType::QUAD)
+            new_elems.push_back(createElement<MeshLib::Quad>(*elem, new_nodes, node_map));
+        else if (elem->getGeomType() == MeshLib::MeshElemType::TETRAHEDRON)
+            new_elems.push_back(createElement<MeshLib::Tet>(*elem, new_nodes, node_map));
+        else if (elem->getGeomType() == MeshLib::MeshElemType::HEXAHEDRON)
+            new_elems.push_back(createElement<MeshLib::Hex>(*elem, new_nodes, node_map));
+        else if (elem->getGeomType() == MeshLib::MeshElemType::PYRAMID)
+            new_elems.push_back(createElement<MeshLib::Pyramid>(*elem, new_nodes, node_map));
+        else if (elem->getGeomType() == MeshLib::MeshElemType::PRISM)
+            new_elems.push_back(createElement<MeshLib::Prism>(*elem, new_nodes, node_map));
+        else
+        {
+            ERR("Error: Unknown element type.");
+            return nullptr;
+        }
+    }
+
+    MeshLib::Properties new_props =
+        constructProperties(mesh.getProperties(), new_elems, node_map, new_nodes.size());
+    return new MeshLib::Mesh("Unity conform mesh", new_nodes, new_elems, new_props);
+}
+
+int main (int argc, char* argv[])
+{
+    ApplicationsLib::LogogSetup logog_setup;
+
+    TCLAP::CmdLine cmd(
+        "Prepares OGS-meshes for use in Unity"
+        "",
+        ' ',
+        "0.1");
+
+    TCLAP::ValueArg<std::string> mesh_arg("i", "input",
+        "the file containing the original OGS mesh", true,
+        "", "input file name");
+    cmd.add(mesh_arg);
+
+    TCLAP::ValueArg<std::string> mesh_out_arg("o", "output",
+        "the file name the result will be written to", true,
+        "", "output file name");
+    cmd.add(mesh_out_arg);
+
+    cmd.parse(argc, argv);
+
+    INFO("Reading mesh \"%s\" ... ", mesh_arg.getValue().c_str());
+    MeshLib::Mesh* mesh = MeshLib::IO::readMeshFromFile(mesh_arg.getValue());
+    if (!mesh)
+        return EXIT_FAILURE;
+    INFO("done.\n");
+
+    INFO("Checking for line elements...")
+    std::array<unsigned, 7> const& n_element_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh);
+    MeshLib::Mesh* result;
+    if (n_element_types[0] == 0)
+    {
+        INFO ("No line elements found.\n");
+        result = mesh;
+    }
+    else if (n_element_types[0] == mesh->getNumberOfElements())
+    {
+        INFO ("Keeping line mesh.\n");
+        result = mesh;
+    }
+    else
+    {
+        MeshLib::ElementSearch searcher(*mesh);
+        std::size_t const n_rem_elems = searcher.searchByElementType(MeshLib::MeshElemType::LINE);
+        result = MeshLib::removeElements(*mesh, searcher.getSearchedElementIDs(), "temp mesh");
+        INFO ("%d line elements found and removed.\n", n_rem_elems);
+    }
+
+    INFO("Checking for cell-arrays...");
+    if (containsCellVecs(*result))
```

Comment:
Just now I opened the PR https://github.com/ufz/ogs/pull/1787 which contains similar functionality. You could use the method `Properties::getPropertyVectorNames(MeshLib::MeshItemType t)` which returns a vector of names of the appropriate `MeshItemType`.  This would save the implementation of `containsCellVecs` and helper functions used inside of `containsVellVecs`.